packages/react/src/components/Modal/Modal.test.tsx

```diff
@@ -89,4 +89,49 @@ describe('Modal', () => {
       },
     );
   });
+
+  it('renders AlertDialog role correctly', () => {
+    // Mock createPortal to capture its component hierarchy in snapshot output.
+    const ReactDOM = require('react-dom');
+    ReactDOM.createPortal = jest.fn(element => {
+      return element;
+    });
+
+    safeCreate(
+      <Modal isOpen={true} isAlert={true} className={'test-className'} containerClassName={'test-containerClassName'}>
+        Test Content
+      </Modal>,
+      component => {
+        expect(component!.toJSON()).toMatchSnapshot();
+
+        ReactDOM.createPortal.mockClear();
+      },
+    );
+  });
+
```

Comment:
Could you add a test for `<Modal isOpen={true} isBlocking={true} isAlert={false}>` that specifically checks for the role to be `dialog`?